symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py

```diff
@@ -39,7 +46,36 @@
     molecules_volume_parameter - bonding_forces_parameter / (units.molar_gas_constant * temperature)
 )
 
-# TODO: Derive from the van der Waals equation of state
+# Derive from the van der Waals equation of state and the virial equation
+
+_volume = compressibility_def.volume
+_mole_count = compressibility_def.amount_of_substance
+_molar_density = SymSymbol("molar_density")
+
+_pressure_expr = solve(vdw_eqn.law, vdw_eqn.pressure)[0].subs({
+    vdw_eqn.volume: _volume,
+    vdw_eqn.temperature: temperature,
+    vdw_eqn.amount_of_substance: _mole_count,
+    vdw_eqn.bonding_forces_parameter: bonding_forces_parameter,
+    vdw_eqn.molecules_volume_parameter: molecules_volume_parameter,
+})
+
+_compressibility_via_volume = compressibility_def.definition.rhs.subs({
+    compressibility_def.pressure: _pressure_expr,
+    compressibility_def.temperature: temperature,
+})
+
+_mole_count_expr = density_qty_law.law.rhs.subs({
+    density_qty_law.volumetric_density: _molar_density,
+    density_qty_law.volume: _volume,
+})
+
+_compressibility_via_density = _compressibility_via_volume.subs(_mole_count, _mole_count_expr).simplify()
+
+# Using the formula for Taylor series coefficients
```

Comment:
Elaborate, how second virial coefficient is related to Taylor series and derivative of compressibility.